src/transaction/boot_sr.c

```diff
@@ -6121,28 +6143,35 @@ boot_dbparm_save_volume (THREAD_ENTRY * thread_p, DB_VOLTYPE voltype, VOLID voli
     }
   else
     {
-      if (boot_Db_parm->temp_nvols < 0 || (boot_Db_parm->temp_nvols == 0 && volid != LOG_MAX_DBVOLID)
-	  || (boot_Db_parm->temp_nvols > 0 && boot_Db_parm->temp_last_volid - 1 != volid))
+      if (boot_Temp_info.temp_nvols < 0 || (boot_Temp_info.temp_nvols == 0 && volid != LOG_MAX_DBVOLID)
+	  || (boot_Temp_info.temp_nvols > 0 && boot_Temp_info.temp_last_volid - 1 != volid))
 	{
 	  /* invalid volid */
 	  assert_release (false);
 	  error_code = ER_FAILED;
 	  goto exit;
 	}
-      boot_Db_parm->temp_nvols++;
-      boot_Db_parm->temp_last_volid = volid;
+      boot_Temp_info.temp_nvols++;
+      boot_Temp_info.temp_last_volid = volid;
     }
 
   /* todo: is flush needed? */
   VPID_GET_FROM_OID (&vpid_boot_bp_parm, boot_Db_parm_oid);
```

Comment:
All this part: logging, update boot_db_parm, heap_flush & fileio_synchronize can be moved to `if (voltype == DB_PERMANENT_VOLTYPE)`

Reply:
Done